### PR TITLE
Bulk cleanup of code in hudson.cli.

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] - Remove unused imports
- [x] - Use diamond operator where possible
- [x] - Use Jenkins#get()

I hit the code issue while I was working on the https://github.com/jenkinsci/jenkins/pull/3144 finalization

No tests needed, No changelogs needed as well

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/cli/declarative/CLIRegisterer.java [hunk 1]
```diff
@@ -146,68 +147,62 @@ private CmdLineParser bindMethod(List<MethodBinder> binders) {
...
-                         *      4 mean IllegalStateException is thrown while performing the command.
-                         *      5 means AbortException is thrown while performing the command.
-                         *      6 means AccessDeniedException is thrown while performing the command.
-                         *      7 means BadCredentialsException is thrown while performing the command.
-                         *      8-15 are reserved for future usage
-                         *      16+ mean a custom CLI exit error code (meaning defined by the CLI command itself)
-                         *
-                         *      <p>
-                         *      Note: For details - see JENKINS-32273
+                         * @param args   Arguments to the sub command. For example, if the CLI is invoked like "java -jar cli.jar foo bar zot",
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/cli/declarative/CLIRegisterer.java]: will revert

File: core/src/main/java/hudson/cli/CLICommand.java [hunk 2]
```diff
@@ -638,7 +638,7 @@ public static CLICommand getCurrent() {
     static {
         // register option handlers that are defined
         ClassLoaders cls = new ClassLoaders();
-        Jenkins j = Jenkins.getActiveInstance();
+        Jenkins j = Jenkins.getInstanceOrNull();
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/cli/CLICommand.java]: I am not sure that using it in `static` initializer is a good idea at all

File: core/src/main/java/hudson/cli/ConnectNodeCommand.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ protected int run() throws Exception {
         boolean errorOccurred = false;
         final Jenkins jenkins = Jenkins.getActiveInstance();
```
Comment (dwnusbaum) [inline core/src/main/java/hudson/cli/ConnectNodeCommand.java]: Is there a reason not to update this (and other usages of `Jenkins#getActiveInstance`) to `Jenkins#get`? Seems like you might as well do it at the same time as updating `Jenkins#getInstance`.

**Review by dwnusbaum — COMMENTED** [review]
This looks good overall, but it might be worth updating usages of `Jenkins#getActiveInstance` at the same time.

**Review by daniel-beck — APPROVED** [review]
LGTM except the reformatted Javadoc

File: core/src/main/java/hudson/cli/handlers/GenericItemOptionHandler.java [hunk 4]
```diff
@@ -57,12 +57,12 @@ protected GenericItemOptionHandler(CmdLineParser parser, OptionDef option, Sette
...
 
     @Override public int parseArguments(Parameters params) throws CmdLineException {
-        final Jenkins j = Jenkins.getInstance();
+        final Jenkins j = Jenkins.get();
         final String src = params.getParameter(0);
         T s = j.getItemByFullName(src, type());
         if (s == null) {
             final Authentication who = Jenkins.getAuthentication();
-            try (ACLContext _ = ACL.as(ACL.SYSTEM)) {
+            try (ACLContext aclContext = ACL.as(ACL.SYSTEM)) {
```
Comment (Wadeck) [inline core/src/main/java/hudson/cli/handlers/GenericItemOptionHandler.java]: Good point since the `_` becomes a reserved keyword.

**Review by Wadeck — APPROVED** [review]
:bee:

File: core/src/main/java/hudson/cli/handlers/GenericItemOptionHandler.java (hunk 4 above)
Comment (oleg-nenashev) [reply core/src/main/java/hudson/cli/handlers/GenericItemOptionHandler.java]: was fixed in another PR

**Comment by oleg-nenashev** [comment]
It should be fine now, fixed the issues and merge conflicts
@Wadeck @daniel-beck would you reapprove it?

**Review by dwnusbaum — APPROVED** [review]

**Review by batmat — APPROVED** [review]

**Comment by jglick** [comment]
Unfortunately produced some merge conflicts with #3838.

**Pull request merged**